app/models/signed_up_team.rb

```diff
@@ -1,4 +1,57 @@
 class SignedUpTeam < ApplicationRecord
-  belongs_to :sign_up_topic
+  belongs_to :project_topic, foreign_key: :sign_up_topic_id
   belongs_to :team
+
+  # Removes all waitlist entries for a given team.
+  # This deletes all sign-up records where the team is waitlisted.
+  def self.drop_off_team_waitlists(team_id)
+    SignedUpTeam.where(team_id: team_id, is_waitlisted: true).destroy_all
+  end
+
+  # Creates a sign-up record for a team by associating the team with the given topic.
+  # This finds the project topic by its ID and then calls the `sign_up_team` method to sign the team up.
+  def self.create_signed_up_team(topic_id, team_id)
+    # Find the project topic by its ID
+    project_topic = ProjectTopic.find(topic_id)
+
+    # Sign up the team for the topic
+    project_topic.sign_up_team(team_id)
+  end
+
+  # Retrieves the team ID for a given user and assignment.
+  # This first finds the team(s) the user is associated with and then retrieves the team for the specified assignment.
+  def self.get_team_id(user_id, assignment_id)
+    # Get the team IDs associated with the given user
+    team_ids = TeamsUser.select('team_id').where(user_id: user_id)
+
+    if team_ids.empty?
+      return nil
+    end
+    # Find the team that matches the assignment ID and retrieve its team_id
+    team_id = Team.where(id: team_ids, assignment_id: assignment_id).first.id
+    # Return the team ID
+    team_id
+  end
+
+  # Deletes all sign-up records for a given team.
+  # This removes all sign-up entries associated with the specified team.
+  def self.drop_off_team_signup_records(team_id)
+    SignedUpTeam.where(team_id: team_id).destroy_all
+  end
+
+  # Reassigns the team to a new topic by removing them from their current topic
+  # and marking them as no longer waitlisted for the new topic.
+  def reassign_topic(topic_id)
```

Comment:
I also think this method could be programmed more clearly, but I am out of time now.